screens/VehicleListScreen.tsx

```diff
@@ -27,46 +28,44 @@ export default function VehicleListScreen({ navigation }) {
   const [data, setData] = useState([]);
   const [count, setCount] = useState(0);
   const [page, setPage] = useState(1);
+  const [nextPageExists, setNextPageExists] = useState(true);
 
-  async function loadNewPage() {
-    const results = await fetchAPI.getVehiclesList(page);
+  async function loadThisPage() {
+    const [results, num_of_results] = await fetchAPI.getVehiclesListAndCount(
+      page
+    );
     setData(data.concat(results));
+    setCount(num_of_results);
+    setNextPageExists(await fetchAPI.nextPageVehiclesExists(page));
     setPage(page + 1);
   }
 
-  async function loadCount() {
-    const no_results = await fetchAPI.getVehiclesCount();
-    setCount(no_results);
-  }
   useEffect(() => {
-    loadCount();
-    loadNewPage();
+    loadThisPage();
     setLoading(false);
   }, []);
 
   const renderItem = ({ item }) => {
-    if (item != undefined)
-      var item_unique = {
-        ...item,
-        id: item.url.split("/")[item.url.split("/").length - 2],
-      };
+    function getId({ item }) {
+      return item.url.split("/")[item.url.split("/").length - 2];
```

Comment:
gdzieś mi kiedyś napisałaś komentarz o url-parse -> faktycznie, tutaj nie zadziała najlepiej. Ale! masz zadanie w osobnym PRku: przenieś tą funkcje (`getId`) do osobnego pliku a następnie napisz do niej unit testy - jak to zrobisz, to wtedy zmienimy jej implementacje na lepszą ;)

(PS: zrób do tego osobnego taska na boardzie)